File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/AddressPersonalMapper.cs
using AddressPersonalSdk = OnlinePayments.Sdk.Domain.AddressPersonal;
using AddressPersonalDto = Business.Domain.Payments.PaymentDetails.AddressPersonal;

namespace Infrastructure.Mappers.PaymentDetails;

public static class AddressPersonalMapper
{
    public static AddressPersonalDto Map(AddressPersonalSdk? response)
    {
        return new()
        {
            AdditionalInfo = response?.AdditionalInfo,
            City = response?.City,
            CountryCode = response?.CountryCode,
            HouseNumber = response?.HouseNumber,
            CompanyName = response?.CompanyName,
            Name = PersonalNameMapper.Map(response?.Name),
            State = response?.State,
            Zip = response?.Zip,
            Street = response?.Street
        };
    }
}

File: server/php/src/Infrastructure/Mappers/PaymentDetails/AddressPersonalMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\AddressPersonal as AddressPersonalDto;
use OnlinePayments\Sdk\Domain\AddressPersonal as AddressPersonalSdk;

final class AddressPersonalMapper
{
    public static function mapFromSdkResponse(?AddressPersonalSdk $response): ?AddressPersonalDto
    {
        if ($response === null) {
            return null;
        }

        return new AddressPersonalDto(
            additionalInfo: $response->additionalInfo ?? null,
            city: $response->city ?? null,
            companyName: $response->companyName ?? null,
            countryCode: $response->countryCode ?? null,
            houseNumber: $response->houseNumber ?? null,
            name: PersonalNameMapper::mapFromSdkResponse($response->name ?? null),
            state: $response->state ?? null,
            street: $response->street ?? null,
            zip: $response->zip ?? null
        );
    }
}
